node/src/actors/session/mod.rs

```diff
@@ -84,6 +84,9 @@ pub struct Session {
 
     /// Reference to config
     config: Arc<Config>,
+
+    /// Expected number of "peers" message from this peer
+    expected_peers_msg: i32,
```

Comment:
I think this cannot be negative, and u8 should be enough

```suggestion
    expected_peers_msg: u8,
```